app/models/course/condition/achievement.rb

```diff
@@ -20,11 +20,12 @@ def satisfied_by?(course_user)
 
   private
 
-  # Given an achievement, returns all its achievement conditions.
+  # Given a conditional object, returns all achievements that it requires.
   #
-  # @param [Course::Achievement] achievement The achievement to return achievement conditions for.
+  # @param [Object] conditional The object that is declared as acts_as_conditional and for which
```

Comment:
Extensions::Conditiona...ConditionalInstanceMethods OR use duck typing. What are the methods each of these classes need to implement, and declare those methods. See the yard documentation.